projects/workflows-creator/src/lib/builder/builder.component.ts

```diff
@@ -202,8 +202,31 @@ export class BuilderComponent<E> implements OnInit, OnChanges {
         });
       });
       this.updateDiagram();
+      this.hideElseBlockIfRequired();
     }
   }
+  /**
+   * This function checks if the else block should be hidden based on the type and number of events in
+   * the event group.
+   */
+  hideElseBlockIfRequired() {
```

Comment:
Here's an improved version of the hideElseBlockIfRequired for better readability:

```javascript
hideElseBlockIfRequired() {
    const events = this.eventGroups[0].children;
    const firstEvent = events[0]?.node;
    
    if (!firstEvent) {
        this.elseBlockHidden = false;
        return;
    }

    let value = firstEvent.state.get('value');

    if (typeof value === 'object') {
        value = value.value;
    }

    const eventType = firstEvent.getIdentifier();
    const eventValue = firstEvent.state.get('value');
    const eventValueType = firstEvent.state.get('valueType');

    this.elseBlockHidden = eventType === EventTypes.OnIntervalEvent ||
        eventType === EventTypes.OnAddItemEvent ||
        (eventType === EventTypes.OnChangeEvent &&
            (eventValue === ValueTypes.AnyValue ||
                eventValueType === ValueTypes.AnyValue));
}
```
changes :

- We check if firstEvent exists before accessing its properties to avoid errors.
- The code is structured to handle cases where firstEvent might be undefined.
- The logic for determining value, eventType, eventValue, and eventValueType remains the same.

This requires testing before implementation, this code improves the readability

Reply:
Done the changes .